hub/constraints/instruction-handling/acc.lisp

```diff
@@ -47,135 +71,171 @@
+(defun    (account-instruction---foreign-code-hash-hi)    (shift  (* account/CODE_HASH_HI account/EXISTS)     ROFF_ACC___ACCOUNT_DOING_ROW  ))
+(defun    (account-instruction---foreign-code-hash-lo)    (shift  (* account/CODE_HASH_LO account/EXISTS)     ROFF_ACC___ACCOUNT_DOING_ROW  ))
+(defun    (account-instruction---current-code-size)       (shift     account/CODE_SIZE                        ROFF_ACC___ACCOUNT_READING_ROW))
+(defun    (account-instruction---current-balance)         (shift     account/BALANCE                          ROFF_ACC___ACCOUNT_READING_ROW))
+
+
+(defconstraint   account-instruction---value-constraints---the-BALANCE-case
+                 (:guard (account-instruction---unexceptional))
+                 ;;
+                 (if-not-zero (account-instruction---is-BALANCE)
+                              (begin
+                                (eq!  (account-instruction---result-hi)    0)
+                                (eq!  (account-instruction---result-lo)    (account-instruction---foreign-balance)))))
+
+(defconstraint   account-instruction---value-constraints---the-EXTCODESIZE-case
+                 (:guard (account-instruction---unexceptional))
+                 ;;
+                 (if-not-zero (account-instruction---is-EXTCODESIZE)
+                              (begin
+                                (eq!   (account-instruction---result-hi)   0)
+                                (eq!   (account-instruction---result-lo)   (account-instruction---foreign-code-size)))))
+
+(defconstraint   account-instruction---value-constraints---the-EXTCODEHASH-case
+                 (:guard (account-instruction---unexceptional))
+                 ;;
+                 (if-not-zero (account-instruction---is-EXTCODEHASH)
+                              (begin
+                                (eq!   (account-instruction---result-hi)   (account-instruction---foreign-code-hash-hi))
+                                (eq!   (account-instruction---result-lo)   (account-instruction---foreign-code-hash-lo)))))
+
+(defconstraint   account-instruction---value-constraints---the-CODESIZE-case
+                 (:guard (account-instruction---unexceptional))
+                 ;;
+                 (if-not-zero (account-instruction---is-CODESIZE)
+                              (begin
+                                (eq!   (account-instruction---result-hi)   0)
+                                (eq!   (account-instruction---result-lo)   (account-instruction---current-code-size)))))
+
+(defconstraint   account-instruction---value-constraints---the-SELFBALANCE-case
+                 (:guard (account-instruction---unexceptional))
+                 ;;
+                 (if-not-zero (account-instruction---is-SELFBALANCE)
+                              (begin
+                                (eq!   (account-instruction---result-hi)   0)
+                                (eq!   (account-instruction---result-lo)   (account-instruction---current-balance)))))
```

Comment:
Splitting of the `account-instruction-value-constraints` constraint and using shorthands to keep things concrete.